### PR TITLE
codelite: upgrade to version 15.0

### PR DESCRIPTION
**Pull request by stahta01**
Replaced, edited or added patch files as needed.
016-mingw-Set-better-target-names.patch helps to
build with wxWidgets version 3.1.5

I am a newbie on using CodeLite; so, I just opened the program and exited it in MinGW64 for testing

### CODE REVIEW

File: mingw-w64-codelite/016-mingw-Set-better-target-names.patch [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+@@ -135,6 +135,15 @@ else (UNIX AND NOT APPLE)
+                           -lz)
+ endif ( UNIX AND NOT APPLE )
+ 
++if(MINGW)
++    set_target_properties(libcodelite
++                          PROPERTIES
++                          PREFIX ""
++                          OUTPUT_NAME libcodelite
++                          RUNTIME_OUTPUT_NAME codelite
```
Comment (lazka) [inline mingw-w64-codelite/016-mingw-Set-better-target-names.patch:27]: any reason why this was changed?

File: mingw-w64-codelite/PKGBUILD [hunk 2]
```diff
@@ -19,59 +20,61 @@ depends=("${MINGW_PACKAGE_PREFIX}-gcc-libs"
...
          )
 optdepends=("${MINGW_PACKAGE_PREFIX}-clang")
 makedepends=("${MINGW_PACKAGE_PREFIX}-boost"
              "${MINGW_PACKAGE_PREFIX}-cmake"
              #"${MINGW_PACKAGE_PREFIX}-clang"
-             "git")
+            )
 options=(strip staticlibs !debug)
+
+# wget --timeout=20 https://github.com/eranif/codelite/archive/15.0.tar.gz
```
Comment (lazka) [inline mingw-w64-codelite/PKGBUILD:33]: remove

File: mingw-w64-codelite/PKGBUILD [hunk 3]
```diff
@@ -19,59 +20,61 @@ depends=("${MINGW_PACKAGE_PREFIX}-gcc-libs"
...
   patch -p1 -i ${srcdir}/003-use-gnu-extensions.patch
   patch -p1 -i ${srcdir}/004-windows-headers-conflict.patch
-  patch -p1 -i ${srcdir}/005-fix-sqlite3-find-and-build.patch
+  patch -p1 -i ${srcdir}/005-cl-15-fix-sqlite3-find-and-build.patch
   patch -p1 -i ${srcdir}/006-proper-libraries-names-mingw.patch
   patch -p1 -i ${srcdir}/007-strip-whitespaces-flags.patch
-  patch -p1 -i ${srcdir}/010-strange-conflicts.patch
-  patch -p1 -i ${srcdir}/011-use-std_string.patch
-  #patch -p1 -i ${srcdir}/012-define-wxCONTROL_NONE.patch
+  #patch -p1 -i ${srcdir}/010-strange-conflicts.patch
```
Comment (lazka) [inline mingw-w64-codelite/PKGBUILD:72]: remove

**Comment by lazka** [comment]
ah, I clicked the merge button... oh well. never mind then :D

File: mingw-w64-codelite/016-mingw-Set-better-target-names.patch (hunk 1 above)
Comment (stahta01) [reply mingw-w64-codelite/016-mingw-Set-better-target-names.patch:27]: To get better results when compiling with wxWidgets 3.1 is why that patch was added.
Those exact lines was adjusted to reduce the number of lines needed and in wxWidgets 3.1
RUNTIME_OUTPUT_NAME libcodelite resulted in files with liblibcodelite in the build log.

Tim S.

File: mingw-w64-codelite/PKGBUILD (hunk 2 above)
Comment (stahta01) [reply mingw-w64-codelite/PKGBUILD:33]: People with bad internet will not like it!

**Pull request merged**